### PR TITLE
statistics: fix incorrect ndv caused by missing fmsketch

### LINKED ISSUE #35044: ndv is incorrect under tidb_analyze_version=1 & tidb_partition_prune_mode=dynamic
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
set global tidb_analyze_version=1;
set tidb_analyze_version=1;
CREATE TABLE `t` (   `id` int(11) DEFAULT NULL,   `a` int(11) DEFAULT NULL, `b` int(11) DEFAULT NULL, `c` int(11) DEFAULT NULL ) PARTITION BY RANGE (`id`) (PARTITION `p0` VALUES LESS THAN (3),  PARTITION `p1` VALUES LESS THAN (7),  PARTITION `p2` VALUES LESS THAN (11));
insert into t values (1,1,1,1),(2,2,2,2),(4,4,4,4),(5,5,5,5),(6,6,6,6),(8,8,8,8),(9,9,9,9);
create index idxa on t (a);
create index idxb on t (b);
create index idxc on t (c);
analyze table t partition p0 index idxa;
analyze table t partition p1 index idxb;
analyze table t partition p2 index idxc;
set global tidb_partition_prune_mode=dynamic;
set tidb_partition_prune_mode=dynamic;
analyze table t partition p0;
show stats_histograms where table_name like "t";
```

### 2. What did you expect to see? (Required)
```
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
| Db_name | Table_name | Partition_name | Column_name | Is_index | Update_time         | Distinct_count | Null_count | Avg_col_size | Correlation |
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
| test    | t          | global         | idxa        |        1 | 2022-05-30 17:50:34 |              7 |          0 |            0 |           0 |
| test    | t          | global         | idxb        |        1 | 2022-05-30 17:50:34 |              7 |          0 |            0 |           0 |
| test    | t          | global         | idxc        |        1 | 2022-05-30 17:50:34 |              7 |          0 |            0 |           0 |
| test    | t          | p0             | idxa        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p0             | idxb        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p0             | idxc        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p1             | idxa        |        1 | 2022-05-30 17:48:30 |              3 |          0 |            0 |           0 |
| test    | t          | p1             | idxb        |        1 | 2022-05-30 17:47:59 |              3 |          0 |            0 |           0 |
| test    | t          | p1             | idxc        |        1 | 2022-05-30 17:48:36 |              3 |          0 |            0 |           0 |
| test    | t          | p2             | idxa        |        1 | 2022-05-30 17:48:30 |              2 |          0 |            0 |           0 |
| test    | t          | p2             | idxb        |        1 | 2022-05-30 17:48:33 |              2 |          0 |            0 |           0 |
| test    | t          | p2             | idxc        |        1 | 2022-05-30 17:48:03 |              2 |          0 |            0 |           0 |
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
12 rows in set (0.00 sec)
```

### 3. What did you see instead (Required)
```
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
| Db_name | Table_name | Partition_name | Column_name | Is_index | Update_time         | Distinct_count | Null_count | Avg_col_size | Correlation |
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
| test    | t          | global         | idxa        |        1 | 2022-05-30 17:50:34 |              7 |          0 |            0 |           0 |
| test    | t          | global         | idxb        |        1 | 2022-05-30 17:50:34 |              4 |          0 |            0 |           0 |
| test    | t          | global         | idxc        |        1 | 2022-05-30 17:50:34 |              5 |          0 |            0 |           0 |
| test    | t          | p0             | idxa        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p0             | idxb        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p0             | idxc        |        1 | 2022-05-30 17:50:34 |              2 |          0 |            0 |           0 |
| test    | t          | p1             | idxa        |        1 | 2022-05-30 17:48:30 |              3 |          0 |            0 |           0 |
| test    | t          | p1             | idxb        |        1 | 2022-05-30 17:47:59 |              3 |          0 |            0 |           0 |
| test    | t          | p1             | idxc        |        1 | 2022-05-30 17:48:36 |              3 |          0 |            0 |           0 |
| test    | t          | p2             | idxa        |        1 | 2022-05-30 17:48:30 |              2 |          0 |            0 |           0 |
| test    | t          | p2             | idxb        |        1 | 2022-05-30 17:48:33 |              2 |          0 |            0 |           0 |
| test    | t          | p2             | idxc        |        1 | 2022-05-30 17:48:03 |              2 |          0 |            0 |           0 |
+---------+------------+----------------+-------------+----------+---------------------+----------------+------------+--------------+-------------+
12 rows in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)
v6.1

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #35044 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

File: executor/analyze_test.go [hunk 1]
```diff
@@ -3316,11 +3316,7 @@ PARTITION BY RANGE ( id ) (
 	tk.MustExec("analyze table t partition p0 index idxa")
 	tk.MustExec("analyze table t partition p1 index idxb")
 	tk.MustExec("set @@session.tidb_partition_prune_mode = 'dynamic'")
-	tk.MustExec("analyze table t partition p0")
-	tk.MustQuery("show warnings").Sort().Check(testkit.Rows(
-		"Warning 8244 Build table: `t` column: `id` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions",
-		"Warning 8244 Build table: `t` index: `idxa` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions",
-	))
+	tk.MustExec("analyze table t partition p0") // no panic
```
Comment (xuyifangreeneyes) [inline executor/analyze_test.go:3319]: Do we need to check warnings here?

**Pull request merged**